cmd/replay/main.go

```diff
@@ -247,7 +247,9 @@ func main() {
 		// Espresso-specific validation
 		// TODO test: https://github.com/EspressoSystems/espresso-sequencer/issues/772
 		isL2Message := message.Message.Header.Kind == arbostypes.L1MessageType_L2Message
-		if isL2Message && chainConfig.Espresso {
+		commitment := espressoTypes.Commitment(wavmio.ReadHotShotCommitment(inboxPos, posInInbox))
+		// When the commitment is not empty, the currently validating message comes from Espresso
```

Comment:
nit: to make this extremely clear, maybe have a separate usingEspresso variable:
```
var validatingAgainstEspresso := commitment != [32]byte{}
```